plugins/inputs/socketstat/socketstat_test.go

```diff
@@ -0,0 +1,123 @@
+				{map[string]interface{}{"state": "ESTAB", "bytes_acked": uint64(19983121), "bytes_received": uint64(266383), "segs_out": uint64(15431), "segs_in": uint64(17633), "data_segs_out": uint64(15119), "data_segs_in": uint64(5098), "recv_q": uint64(0), "send_q": uint64(0)}},
+			},
+		},
+		{
+			name:     "udp packets captured",
+			proto:    []string{"udp"},
+			filename: "udp_traffic.txt",
+			tags: []map[string]string{
+				{"proto": "udp", "local_addr": "10.10.0.4", "local_port": "33149", "remote_addr": "10.10.0.5", "remote_port": "53"},
+				{"proto": "udp", "local_addr": "10.10.0.4", "local_port": "54276", "remote_addr": "10.10.0.6", "remote_port": "53"},
+				{"proto": "udp", "local_addr": "10.10.0.4", "local_port": "38312", "remote_addr": "10.10.0.7", "remote_port": "53"},
+			},
+			fields: [][]map[string]interface{}{
+				{map[string]interface{}{"recv_q": uint64(0), "send_q": uint64(0)}},
+				{map[string]interface{}{"recv_q": uint64(0), "send_q": uint64(0)}},
+				{map[string]interface{}{"recv_q": uint64(0), "send_q": uint64(0)}},
+			},
+		},
+	}
+	for i, tt := range tests {
+		octets, err := os.ReadFile(filepath.Join("testdata", tt.filename))
+		require.NoError(t, err)
+
+		t.Run(tt.name, func(t *testing.T) {
+			i++
+			ss := &Socketstat{
+				SocketProto: tt.proto,
+			}
+			acc := new(testutil.Accumulator)
+
+			err := ss.Init()
+			if err != nil && !strings.Contains(err.Error(), "executable file not found in $PATH") {
+				require.NoError(t, err)
+			}
+			ss.lister = func(cmdName string, proto string, timeout config.Duration) (*bytes.Buffer, error) {
+				return bytes.NewBuffer(octets), nil
+			}
+
+			err = acc.GatherError(ss.Gather)
+			require.ErrorIs(t, err, tt.err)
+			if len(tt.proto) == 0 {
+				n := acc.NFields()
+				require.Equalf(t, 0, n, "%d: expected 0 values got %d", i, n)
+				return
+			}
+			if len(tt.tags) == 0 {
+				n := acc.NFields()
+				require.Equalf(t, 0, n, "%d: expected 0 values got %d", i, n)
+				return
+			}
+			n := 0
+			for j, tags := range tt.tags {
+				for k, fields := range tt.fields[j] {
+					require.Greater(t, len(acc.Metrics), n)
+					m := acc.Metrics[n]
+					require.Equal(t, measurement, m.Measurement, "%d %d %d: expected measurement '%#v' got '%#v'\n", i, j, k, measurement, m.Measurement)
+					require.Equal(t, tags, m.Tags, "%d %d %d: expected tags\n%#v got\n%#v\n", i, j, k, tags, m.Tags)
+					require.Equal(t, fields, m.Fields, "%d %d %d: expected fields\n%#v got\n%#v\n", i, j, k, fields, m.Fields)
+					n++
+				}
+			}
+		})
+	}
+}
+
+func TestSocketstat_Gather_listerError(t *testing.T) {
+	errFoo := errors.New("error foobar")
+	ss := &Socketstat{
+		SocketProto: []string{"foobar"},
+	}
+	ss.lister = func(cmdName string, proto string, timeout config.Duration) (*bytes.Buffer, error) {
+		return new(bytes.Buffer), errFoo
+	}
+	acc := new(testutil.Accumulator)
+	err := acc.GatherError(ss.Gather)
+	require.ErrorIs(t, errFoo, err)
```

Comment:
Please use `EqualError` here
```suggestion
	ss := &Socketstat{
		SocketProto: []string{"foobar"},
	}
	ss.lister = func(cmdName string, proto string, timeout config.Duration) (*bytes.Buffer, error) {
		return new(bytes.Buffer), errFoo
	}
	acc := new(testutil.Accumulator)
	err := acc.GatherError(ss.Gather)
	require.EqualErrorIs(t, err, "error foobar")
```

Reply:
> @sajoupa this is almost perfect. Just a few small things left. I tried to make suggestions in the hope to make this the last iteration. I think we are almost there.

No worries at all, I'm not in a rush to get this plugin live, and it's never too late to make better code anyway. Thanks for all the reviews !

Reply:
Pushed a fix (using a variable for "error foobar")